File: app/build-app2/tmp/kapt3/stubs/debug/eu/villacristina/lechugapro/ui/comercializacion/ingreso/IngresoListaViewModel.java
package eu.villacristina.lechugapro.ui.comercializacion.ingreso;

@kotlin.Metadata(mv = {1, 9, 0}, k = 1, xi = 48, d1 = {"\u0000*\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\u0010 \n\u0002\u0018\u0002\n\u0002\b\u0003\n\u0002\u0018\u0002\n\u0002\b\u0003\u0018\u00002\u00020\u0001B\r\u0012\u0006\u0010\u0002\u001a\u00020\u0003\u00a2\u0006\u0002\u0010\u0004J\u000e\u0010\u000b\u001a\u00020\f2\u0006\u0010\r\u001a\u00020\bJ\u000e\u0010\u000e\u001a\u00020\f2\u0006\u0010\r\u001a\u00020\bR\u000e\u0010\u0002\u001a\u00020\u0003X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u001d\u0010\u0005\u001a\u000e\u0012\n\u0012\b\u0012\u0004\u0012\u00020\b0\u00070\u0006\u00a2\u0006\b\n\u0000\u001a\u0004\b\t\u0010\n\u00a8\u0006\u000f"}, d2 = {"Leu/villacristina/lechugapro/ui/comercializacion/ingreso/IngresoListaViewModel;", "Landroidx/lifecycle/ViewModel;", "repository", "Leu/villacristina/lechugapro/data/IngresoRepositoryContract;", "(Leu/villacristina/lechugapro/data/IngresoRepositoryContract;)V", "todosLosIngresos", "Lkotlinx/coroutines/flow/StateFlow;", "", "Leu/villacristina/lechugapro/data/Ingreso;", "getTodosLosIngresos", "()Lkotlinx/coroutines/flow/StateFlow;", "deleteIngreso", "Lkotlinx/coroutines/Job;", "ingreso", "reInsertIngreso", "app_debug"})
public final class IngresoListaViewModel extends androidx.lifecycle.ViewModel {
    @org.jetbrains.annotations.NotNull()
    private final eu.villacristina.lechugapro.data.IngresoRepositoryContract repository = null;
    @org.jetbrains.annotations.NotNull()
    private final kotlinx.coroutines.flow.StateFlow<java.util.List<eu.villacristina.lechugapro.data.Ingreso>> todosLosIngresos = null;
    
    public IngresoListaViewModel(@org.jetbrains.annotations.NotNull()
    eu.villacristina.lechugapro.data.IngresoRepositoryContract repository) {
        super();
    }
    
    @org.jetbrains.annotations.NotNull()
    public final kotlinx.coroutines.flow.StateFlow<java.util.List<eu.villacristina.lechugapro.data.Ingreso>> getTodosLosIngresos() {
        return null;
    }
    
    @org.jetbrains.annotations.NotNull()
    public final kotlinx.coroutines.Job deleteIngreso(@org.jetbrains.annotations.NotNull()
    eu.villacristina.lechugapro.data.Ingreso ingreso) {
        return null;
    }
    
    @org.jetbrains.annotations.NotNull()
    public final kotlinx.coroutines.Job reInsertIngreso(@org.jetbrains.annotations.NotNull()
    eu.villacristina.lechugapro.data.Ingreso ingreso) {
        return null;
    }
}

File: app/src/main/java/eu/villacristina/lechugapro/ui/comercializacion/ingreso/IngresoListaViewModel.kt
package eu.villacristina.lechugapro.ui.comercializacion.ingreso

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.viewModelScope
import eu.villacristina.lechugapro.data.Ingreso
import eu.villacristina.lechugapro.data.IngresoRepository
import eu.villacristina.lechugapro.data.IngresoRepositoryContract
import kotlinx.coroutines.flow.SharingStarted
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.stateIn
import kotlinx.coroutines.launch

class IngresoListaViewModel(private val repository: IngresoRepositoryContract) : ViewModel() {

    val todosLosIngresos: StateFlow<List<Ingreso>> = repository.todosLosIngresos
        .stateIn(
            scope = viewModelScope,
            started = SharingStarted.WhileSubscribed(5000),
            initialValue = emptyList()
        )

    fun deleteIngreso(ingreso: Ingreso) = viewModelScope.launch {
        repository.delete(ingreso)
    }

    fun reInsertIngreso(ingreso: Ingreso) = viewModelScope.launch {
        repository.insert(ingreso)
    }
}

class IngresoListaViewModelFactory(private val repository: IngresoRepositoryContract) : ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(IngresoListaViewModel::class.java)) {
            @Suppress("UNCHECKED_CAST")
            return IngresoListaViewModel(repository) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }
}
